### DISCUSSION TITLE
2023-09-15 - Testing 4 - Do you get a signature before starting work? - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#6703] [Chewing the Fat]
Hey SSW'ers,

Let's talk about this rule | [Do you ask clients to approve your specifications?](https://www.ssw.com.au/rules/ask-clients-approval/)

**Comment by Dhruv-0987** [comment]
This is a concept I learnt after joining SSW there so much stuff involved when making a software for clients and covering for yourself is the biggest one, this rule gives sure shot way of doing that.

**Comment by PennyWalker** [comment]
I'm actually a real fan of the checked by, I think it makes me a better tester - when you have to put your name to something, you pay much closer attention and double check everything before signing off. I can see how this would be really beneficial for things like mock ups - I've used it with Adam before to approve design versions etc, and it helps back you up later if you encounter arguments over what was agreed.

**Comment by JeanThirion** [comment]
One my favourite SSW rules, most likely something that will follow me for a while. Simple and effective ✅

**Comment by matt-goldman** [comment]
Good rule. Anything we can do to get people we are working with, especially clients, to take accountability is worth its weight in gold.

**Comment by bradystroud** [comment]
Good stuff, makes sense 👍🏼

I don't really get the MJ metaphor but I can't think of anything better.
Also dunno what this has to do with testing

**Comment by ozdemir-mehmet** [comment]
A practical rule that everyone can follow.

**Comment by tkapa** [comment]
Easy, fundamental rule. When I think of SSW Rules, this is what comes to mind.

**Comment by 2fernandez** [comment]
❤️ The Rule - I love the concept behind the rule
✅ Video - Overall, a great effort by the video team
✅ Video - It's simple and clean; I love it when they use a fill light so the talent has more colour to his face

❌  Video  - I constantly see the footage moving out of focus a fair bit, especially at the start 
❌  Video - The video seems to jump around the 0:19 / 0:21 mark; it is not smooth.
❌  Video - When zooming in or panning, it’s not a great idea to do this in post-production; Pulling off this effect effectively is challenging. Please use a physical slider and not rely on 4k imagery as a tool.

❌  The Rule - The Michael Jackson image should be moved to the bottom of the rule and there should only be 1 image signed image. "Less is more"

**Reply by drwharris** [reply]
The zooming in and out I noticed to. not cool.

**Comment by bahjatalaadelssw** [comment]
This is a simple way to manage expectations and accountability.

**Comment by leahy268** [comment]
Huge fan of this rule.  I think when you are young you get caught once or twice doing a job and having to redo the job for free because the client didn't have the same understanding as you.  On occasion I think they had just changed their mind.  Then you start to just ask for them to sign off on it.  Either on paper or in an email and then when they change their mind you can show them what they agreed to.

**Comment by leehawkinsssw** [comment]
I get the idea, essentially a CYA and a way to nudge stakeholders to think more deeply about their statements of what they want us to implement. I don't see this as anything to do with testing par se.

The rule only needs one MJ image with signature, not two. I'm not convinced that referencing MJ at all is a great idea any more.

**Comment by UlyssesMaclaren** [comment]
The tips and explanation were great, and this is an important rule. 

I recommend removing the images and references to Michael Jackson on the rule to make it more likely to be shared and used by others. There are strong negative connotations around MJ, and I would never share this rule because of it, which is unfortunate and unnecessary. (Ie MJ being in the rule does nothing to help explain it)

**Reply by leehawkinsssw** [reply]
+1 on removing the MJ references, the point of the rule is made without referencing MJ.

**Comment by MarcusIrmscherSSW** [comment]
Great rule, helps avoid any confusion and unrealistic expectations on delivery.

**Comment by MattParkerDev** [comment]
Loved the video from Adam, and an essential Rule.

**Comment by Ben0189** [comment]
This is a great rules, but i don't understand why it is a Michael Jackson rules

**Reply by Levijj22** [reply]
It was notoriously difficult to get his signature. Especially now...

**Comment by joshbermanssw** [comment]
Great and effective practice

**Comment by ricksu978** [comment]
MJ's rule is great; it really helps clients engage more deeply with the content.

**Comment by jakebayliss** [comment]
Great rule - shares responsibility for what is developed with the product owner

**Comment by apforsyth** [comment]
Some clients wont sign off but you do the work anyway, and document their changing brief as you go, keep them up to date re estimated time and cost.  Not ideal but sometimes it happens!

**Comment by zacharykeeping** [comment]
This is a good and simple rule. I agree that it's probably best to keep MJ out of it however.

**Comment by vladislav-kir** [comment]
This is a great rule, and the video was very good :)

**Comment by AshAnil** [comment]
Nice rule ⭐ "Checked by" process always works and adding a signature to sign off is a good practice to make sure the client is in sync with us.

**Comment by chrisschultzssw** [comment]
Great rule - it's important to confirm there is an understanding with the client on the work that will be done before starting.

**Comment by jackreimers** [comment]
Great rule, gives a lot of context to why we do checked by on our emails, such a small action can save so much trouble later on in a project.

**Comment by Jord-Gui** [comment]
Great rule to ensure everyone is on the same page before work commences.

**Comment by AlexRothwell** [comment]
I really liked the presentation in this video, it was clear and to the point. It's an important part of the process when delivering a project for a client, saving everyone time and ensuring alignment on the goal

I do however agree with many others that the Rule & video would be easier to share and stronger without the MJ references - I can't even really identify how he is related anyway

**Comment by jomelsontanco** [comment]
Usually client pushbacks I've encountered when asking them to sign something comes as a light-hearted jest. 
These half-meant jokes have always annoyed me because they think I'm doing it to cover myself which in reality I'm doing it for their best interests. 
I love this rule because it is only one that I have read so far who prescribes a simple way of explaining why something needs to be signed by a client.

**Comment by piers-sinclair** [comment]
It's a great way to force the Product Owner to pay extra close attention to detail.

**Comment by jontygardner** [comment]
This is a great way to subtly ensure the Product Owner takes accountability for approving work! 

[![gif](https://github.com/SSWConsulting/SSW.Rules.Content/assets/97415708/ab3cec16-1a4e-4435-ae4a-c9e10b1dd4b8)](https://tenor.com/bl079.gif)

**Comment by tomek-i** [comment]
Many times we write in a way that makes sense to us, which does not necessarily make sense to the person the email is addressed to.
Going through the content before sending it off and getting a "checked by"  from the receiver makes sure that both ends are in sync and allows to make amendments to the email and make sure both ends understand and agree on the content.

**Comment by sethdaily** [comment]
Foundational principle.

**Comment by YalalCakirSSW** [comment]
Great rule, clears off any ambiguity.

**Comment by SamBWagner** [comment]
I really like this rule. I've just found it easier to employ the "as per my conversation" rule in it's place. Feels more natural

**Comment by amankumarrr** [comment]
Over time, I've come to appreciate the importance of this rule. When I follow this practice of requesting people to review my emails, they tend to be more engaged in the process, providing valuable feedback and insights.

**Comment by steven0x51** [comment]
Great rule, saves later arguments. but why is it called the MJ rule?

**Comment by jernejk** [comment]
I learnt this rule very long ago and what I have learned.
It's very good advice, just make sure the client doesn't agree and sign contradictory statements or multiple versions because then you're in a world of pain. 😅

I prefer to get everything in PBIs and get them checked there because we have a track record when it checks all of the changes made and they can check a more fleshed-out work as emails don't always translate well into PBIs.

Depending on the client, some prefer to do this before you start work, some prefer at the very beginning of the project and others prefer to do this occasionally with everything done in bulk.

If you struggle with this, just imagine you are working for a law firm that offers software development and every PBI is a contract. Helps me with clients who hate this.

**Comment by jeoffreyfischer** [comment]
Very useful little tricks, will definitely use them :)

**Comment by NadeekaJK** [comment]
Love the rule and MJ 💗
but I don't  get where MJ fits in to this rule.. Did he not get a checked by from his manager before he gave Bob his signature?? or from Jackson five?? or from his Dad??? 🤯

**Comment by JackDuan123** [comment]
Asking approval is what I do all the time,
The hardship is getting the clinet to understand the value of task or getting them to understand a task has no value at all.

**Comment by stephanfakossw** [comment]
Great rule that makes sure we're all in sync.

**Comment by AttackOnMorty** [comment]
My first impression when I saw today’s CTF was: Is this too formal?
Later: It makes sense.

**Comment by christoment** [comment]
Very good rule, I think this works as a prevention mechanism and also as the CYA rule for us consultants :)

**Comment by danielmackay** [comment]
I like the idea behind this rule, and the extra attention someone will usually pay before putting their name against something.  It's also made me conscious of asking for approval on content that is linked elsewhere (e.g design document, PBI AC's etc).

However, I wonder within most issue tracking systems, if we could still link the PBI in an email as long is there is sufficient audit tracking of the PBI history?

**Comment by Harry-Ross** [comment]
I think this is a great rule that reminds us the importance of getting written accountability from clients. I'll be sure to use it!

**Comment by CoffeeBugHub** [comment]
A good rule that makes devs aware that how important is to get client's written approval or signature especially for client projects.

**Comment by JackDevAU** [comment]
Checked by is great - one of the most important rules for client accountability and to cover our back! Good rule and video Adam!

**Comment by AnkyChe** [comment]
Great rule and video. Will keep it in mind and use it.

**Comment by jimmidier** [comment]
Impressive rule and speech, the idea and methodology behind is enlightening for dealing with other business activities.

**Comment by KristenHu** [comment]
Great and practical rule! I'll definitely use it, but I wonder why it is called MJ rule?

**Comment by Aibono1225** [comment]
One of my favorite SSW rules. The 'checked by' really adds value to the work.

**Comment by babakamyljanovssw** [comment]
Great rule! It is good practice to have 'checked by' on PBIs to avoid further complications.

**Comment by johnxu16** [comment]
It's imaginable that how the inventor of the rules been tortured. Very inspirating rule!

**Comment by jerryluossw** [comment]
Great rule, but it doesn't work well for China clients, they're willing to admit the content of the email, but it's not easy to ask them to sign (Add check by) on email 😮‍💨

**Comment by tino-liu** [comment]
You checked it! No backsies! Simple, elegant, and effective.

**Comment by SofieHong** [comment]
"Checked by" is one of my favourite rules, it can effectively avoid the problem of misunderstandings

**Comment by tombui99** [comment]
This is definitely the most fundamental and useful rule that I learnt when I first joined SSW and have been following it ever since. It helps to make sure that everyone is in sync and reduce waste time!

**Comment by Levijj22** [comment]
Great video! 
A unique concept, one that takes time to get used to. 
So many benefits though!

**Comment by drwharris** [comment]
For me on CER this takes the form of an email combining the "Checked by" and "As per our Conversation" rules.

**Comment by BrookJeynes** [comment]
Very important rule to ensure things are agreed on properly and to get the PO (or others) more attentive to what they're agreeing on

**Comment by Marxoz** [comment]
The checked by and as per our conversation is gold. But tbh, I have never used it inside a PBI. I do get the PBI's checked though, but it makes sense to add it to the PBI to ensure the PO / Stakeholder is really trying to understand the content.

**Comment by Abf09** [comment]
I see the importance of this rule, I wouldn't be happy if the client says in the end "This isn't what I wanted" considering they agreed to it earlier. This is something that saves a lot of pain for the developers and the client.

**Comment by TobyChurches** [comment]
Great Rule, Definitely helps to protect yourself further down the track.

**Comment by cvionnet** [comment]
Great video and rule!
The tip about adding screenshots of the content instead of a link is valuable.
